OniProj/OniGameSource/Oni_Object/Oni_Object.h

```diff
@@ -820,7 +821,7 @@ typedef struct OBJtOSD_Combat
 	UUtUns16				id;
 	UUtUns16				flags;
 	float					medium_range;
-	UUtUns32				behavior[5];
+	UUtUns32				behavior[AI2cCombatRange_NumStoredRanges];
```

Comment:
Is this struct ever (de)serialized?
To remain compatible with saves/data.

Reply:
This guy has been 5 before and after; The array in `AI2tCombatSettings` has changed though. I didn't find code that serializes AI state.
It looks like [`AIiScript_SaveGame()`](https://github.com/hogsy/OniFoxed/blob/9f8c643a6d54d6032a7da72fafc803bf4fb806a2/OniProj/OniGameSource/Oni_AI/Oni_AI_Script.c#L1395) is the function that is called to save the game, which calls [`ONrGameState_MakeContinue()`](https://github.com/hogsy/OniFoxed/blob/9f8c643a6d54d6032a7da72fafc803bf4fb806a2/OniProj/OniGameSource/Oni_GameState.c#L6625). That stores some data in `ONgContinue`, forwards to [`ONrPersist_SetContinue()`](https://github.com/hogsy/OniFoxed/blob/9bbe1f4cc5a6d8fcfed2d1b83f9b990513bf434a/OniProj/OniGameSource/Oni_Persistance.c#L348) which copies that into `ONgPersistance.continues`, which is then written to `persist.dat` via [`ONrPersist()`](https://github.com/hogsy/OniFoxed/blob/9bbe1f4cc5a6d8fcfed2d1b83f9b990513bf434a/OniProj/OniGameSource/Oni_Persistance.c#L152).

Reply:
I think you're right.